datastore/tests/unit/test_client.py

```diff
@@ -860,13 +860,43 @@ def test_allocate_ids_w_partial_key(self):
         # Check the IDs returned.
         self.assertEqual([key._id for key in result], list(range(num_ids)))
 
-    def test_allocate_ids_with_completed_key(self):
+    def test_allocate_ids_w_completed_key(self):
         creds = _make_credentials()
         client = self._make_one(credentials=creds)
 
         COMPLETE_KEY = _Key(self.PROJECT)
         self.assertRaises(ValueError, client.allocate_ids, COMPLETE_KEY, 2)
 
+    def test_reserve_ids_w_completed_key(self):
+        num_ids = 2
+        creds = _make_credentials()
+        client = self._make_one(credentials=creds,_use_grpc=False)
+        COMPLETE_KEY = _Key(self.PROJECT)
+        alloc_ids = mock.Mock(return_value=None, spec=[])
```

Comment:
Returning `None` here is incorrect:  the mock should return another object to emulate the `ReserveIdsResponse` message that the API actually returns.

Reply:
This test case is regarding to the datastore.client class  , we expose method reserve_ids for that and it returns None as you commented before, datastore_v1.DatastoreClient's method reserve_ids returns the object ReserveIdsResponse,  So not getting it why to mock ReserveIdsResponse message when it is empty  and how i compare with the result with return by datastore.client method ,so may i remove return_value=None from mock and leave blank like reserve_ids = mock.Mock(spec=[])

Reply:
Yes, you can just let the mock have its default return value.  The test would then fail (because the caller is returning `response_pb`), which we then fix by explicitly returning `None`.